processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -536,6 +541,9 @@ TimeAndDims toTimeAndDims(InputRow row) throws IndexSizeExceededException
             columnCapabilities.put(dimension, capabilities);
           }
           DimensionHandler handler = DimensionHandlerUtil.getHandlerFromCapabilities(dimension, capabilities);
+          if (handler == null) {
```

Comment:
In most places getHandlerFromCapabilities() is called, you create a StringDimensionHandler if the handler is null. Can you just make getHandlerFromCapabilities() never return null and return a StringDimensionHandler by default or would that break something?